exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -110,6 +118,48 @@ public Viewable submitQuery(@FormParam("query") String query,
     }
   }
 
+  @Nonnull
+  private Map<String, Object> readOptionsFromForm(Form form) {
+    Map<String, Object> options = new HashMap<String, Object>();
+    SessionOptionManager sessionOptionManager = webUserConnection.getSession().getOptions();
+    for (Map.Entry<String, List<String>> pair : form.asMap().entrySet()) {
+      String name = pair.getKey();
+      OptionDefinition definition = sessionOptionManager.getOptionDefinition(name);
+      if (definition != null) {
+        String valueStr = pair.getValue().get(0);
+        if (!valueStr.isEmpty()) {
+          try {
+            Object value = null;
+            switch (definition.getValidator().getKind()) {
```

Comment:
You do iteration over switch by kind twice here and in  `QueryWrapper` class, I think you can do only once in QueryWrapper.

Reply:
When I parse the form the inputs are all strings, but when the input comes from JSON they can be mixed types.  I think it is better this way despite the apparent duplication.